libs/generic-view/store/src/lib/entities/delete-entities-data.action.ts

```diff
@@ -10,31 +10,45 @@ import { ReduxRootState } from "Core/__deprecated__/renderer/store"
 import { ActionName } from "../action-names"
 import { EntityId } from "device/models"
 import { difference } from "lodash"
+import delayResponse from "@appnroll/delay-response"
+
+interface DeleteEntitiesDataActionPayload {
+  entitiesType: string
+  ids: EntityId[]
+  deviceId: DeviceId
+  onSuccess?: () => Promise<void> | void
+  onError?: () => Promise<void> | void
+}
 
 export const deleteEntitiesDataAction = createAsyncThunk<
   EntityId[],
-  {
-    entitiesType: string
-    ids: EntityId[]
-    deviceId: DeviceId
-  },
+  DeleteEntitiesDataActionPayload,
   { state: ReduxRootState }
 >(
   ActionName.DeleteEntityData,
-  async ({ entitiesType, ids, deviceId }, { rejectWithValue }) => {
-    const response = await deleteEntitiesDataRequest({
-      entitiesType,
-      ids,
-      deviceId,
-    })
-
+  async (
+    { entitiesType, ids, deviceId, onSuccess, onError },
+    { rejectWithValue }
+  ) => {
+    const response = await delayResponse(
```

Comment:
Only the response is delayed, not the request itself. In fact it forces the action to perform for at least one second. This is needed to avoid showing "deleting progress" modal for fraction of a second which could be considered a UI glitch.